WhateverGreen/kern_igfx.hpp

```diff
@@ -457,7 +492,668 @@ class IGFX {
 	/**
 	 *  ReadAUX wrapper to modify the maximum link rate value in the DPCD buffer
 	 */
-	static int wrapReadAUX(void *that, IORegistryEntry *framebuffer, uint32_t address, uint16_t length, void *buffer, void *displayPath);
+	static IOReturn wrapReadAUX(void *that, IORegistryEntry *framebuffer, uint32_t address, uint16_t length, void *buffer, void *displayPath);
+	
+	/**
+	 *	Reflect the `AppleIntelFramebufferController::CRTCParams` struct
+	 *
+	 *	@note Unlike the Intel Linux Graphics Driver,
+	 *	- Apple does not transform the `pdiv`, `qdiv` and `kdiv` fields.
+	 *	- Apple records the final central frequency divided by 15625.
+	 *	@ref static void skl_wrpll_params_populate(params:afe_clock:central_freq:p0:p1:p2:)
+	 *	@seealso https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/gpu/drm/i915/intel_dpll_mgr.c?h=v5.1.13#n1171
+	 */
+	struct CRTCParams
+	{
+		/// Uninvestigated fields
+		uint8_t uninvestigated[32];
+		
+		/// P0                          [`CRTCParams` field offset 0x20]
+		uint32_t pdiv;
+		
+		/// P1                          [`CRTCParams` field offset 0x24]
+		uint32_t qdiv;
+		
+		/// P2                          [`CRTCParams` field offset 0x28]
+		uint32_t kdiv;
+		
+		/// Difference in Hz            [`CRTCParams` field offset 0x2C]
+		uint32_t fraction;
+		
+		/// Multiplier of 24 MHz        [`CRTCParams` field offset 0x30]
+		uint32_t multiplier;
+		
+		/// Central Frequency / 15625   [`CRTCParams` field offset 0x34]
+		uint32_t cf15625;
+		
+		/// Other fields that are not of interest
+		uint8_t others[0];
+	};
+	
+	/**
+	 *	Represents the current context of probing dividers for HDMI connections
+	 */
+	struct ProbeContext
+	{
```

Comment:
Might you want to put { to `struct ProbeContext`? I.e. `struct ProbeContext {` This is what the rest of the code does. The same issue exists with CRTCParams, DisplayPortDualModeAdapterInfo, and LSPCON.